lms/models/notes.py

```diff
@@ -0,0 +1,58 @@
+from typing import Optional
+
+from flask import jsonify, request
+from flask_login import current_user
+
+from lms.lmsdb.models import CommentText, Exercise, Note, NotePrivacy, User
+from lms.models.errors import fail
+
+
+def create_note_instance(
+    creator: User,
+    user: User,
+    note_text: CommentText,
+    privacy: NotePrivacy,
+    exercise: Optional[Exercise] = None,
+) -> Note:
+    return Note.get_or_create(
+        creator=creator,
+        user=user,
+        note=note_text,
+        exercise=exercise,
+        privacy=privacy,
+    )
+
+
+def delete_note():
+    note_id = int(request.args.get('noteId'))
+    note_ = Note.get_or_none(Note.id == note_id)
+    if (
+        note_.creator.id != current_user.id
+        and note_.is_private
+    ):
+        return fail(403, "You aren't allowed to access this page.")
+    if note_ is not None:
+        note_.delete_instance()
+    return jsonify({'success': 'true'})
+
+
+def create_note(user: User, user_id: int):
```

Comment:
This one doesn't belong into here - `models` should receive primitive parameters, process things in a databasish way and return primitive values.